src/blueapi/service/interface.py

```diff
@@ -20,52 +18,81 @@
 context and worker"""
 
 
-class _Singleton:
-    context: BlueskyContext
-    worker: Worker
-    messaging_template: MessagingTemplate | None = None
-    initialized = False
+@runtime_checkable
+class CacheClearable(Protocol):
+    def cache_clear(self) -> None: ...
 
 
-def start_worker(
-    config: ApplicationConfig,
-    bluesky_context: BlueskyContext = None,
-    worker: TaskWorker = None,
-) -> None:
-    """Creates and starts a worker with supplied config"""
-    if _Singleton.initialized:
-        raise Exception("Worker is already running. To reload call stop first")
-    _Singleton.context = bluesky_context
-    if _Singleton.context is None:
-        _Singleton.context = BlueskyContext()
-        _Singleton.context.with_config(config.env)
-    _Singleton.worker = worker
-    if _Singleton.worker is None:
-        _Singleton.worker = TaskWorker(
-            _Singleton.context,
-            broadcast_statuses=config.env.events.broadcast_status_events,
-        )
-    if config.stomp is not None:
-        _Singleton.messaging_template = StompMessagingTemplate.autoconfigured(
-            config.stomp
-        )
+_CONFIG: ApplicationConfig = ApplicationConfig()
 
-    # Start worker and setup events
-    _Singleton.worker.start()
-    if _Singleton.messaging_template is not None:
-        event_topic = _Singleton.messaging_template.destinations.topic(
-            "public.worker.event"
-        )
+
+def config() -> ApplicationConfig:
+    return _CONFIG
+
+
+def set_config(new_config: ApplicationConfig):
+    global _CONFIG
+
+    _CONFIG = new_config
+
+
+@lru_cache
```

Comment:
```suggestion
@lru_cache(maxsize=1)
```

Is this clearer for what the intended behaviour is?

Reply:
Possibly although that parameter seems to be more relevant when the function has parameters. I think in this case it's ok as is.